### ISSUE TITLE
Add cgroup creation details to `cluster_spec.md` for Compute Resources Reservation

### ISSUE THREAD

**Issue by minight** [kubernetes/kops#12324]
I noticed while attempting to set Compute Resources Reservations (https://github.com/kubernetes/kops/blob/master/docs/cluster_spec.md#compute-resources-reservation) on our kops cluster on AWS, that our nodes did not have the `kube-reserved` and `system-reserved` cgroups created on startup, resulting in the kubelet failing to start when `enforceNodeAllocatable` is set to `pods,system-reserved,kube-reserved`

The kubernetes documentation states `Note that Kubelet does not create --kube-reserved-cgroup if it doesn't exist. Kubelet will fail if an invalid cgroup is specified.` (https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#kube-reserved) in the linked section.

I propose adding some details to the documentation outlining how to provision cgroups on node start-up using cloud-init, thus permitting the kubelet to start.

This should assist in kops users setting up reserved cgroups for the kubelet/containerd for the first time when reading the documentation.

# Suggested Section

To the end of the `##  Compute Resources Reservation` section (https://github.com/kubernetes/kops/blob/master/docs/cluster_spec.md#compute-resources-reservation) , I propose adding:

This assumes that the `kube-reserved` and `system-reserved` cgroups exist on the node prior to kubelet start. Depending on your node image, these may not exist and hence prevent the kubelet from starting. To ensure these exist, you can provision these cgroups with the following additions to each `InstanceGroup` spec:

```
kind: InstanceGroup
...
spec:
  fileAssets:
    - name: system-reserved-slice
      path: /etc/systemd/system/system-reserved.slice
      content: |
        [Unit]
        Description=Limited resources slice for System services
        Documentation=man:systemd.special(7)
        DefaultDependencies=no
        Before=slices.target
        Requires=-.slice
        After=-.slice
    - name: kube-reserved-slice
      path: /etc/systemd/system/kube-reserved.slice
      content: |
        [Unit]
        Description=Limited resources slice for Kubernetes services
        Documentation=man:systemd.special(7)
        DefaultDependencies=no
        Before=slices.target
        Requires=-.slice
        After=-.slice
    - name: kubelet-cgroup
      path: /etc/systemd/system/kubelet.service.d/10-cgroup.conf
      content: |
        [Service]
        Slice=kube-reserved.slice
    - name: containerd-cgroup
      path: /etc/systemd/system/containerd.service.d/10-cgroup.conf
      content: |
        [Service]
        Slice=kube-reserved.slice
```

# Attribution

The snippet above is derived off the work of Kashif Saadat
that I found at https://medium.com/appvia/prevent-resource-starvation-of-critical-system-and-kubernetes-services-83cfeaa6ac96

**Comment by olemarkus** [comment]
I think this looks good. At least until kops can create these cgroups. Can you do a PR? It's just MD files in the `docs` folder.

**Comment by olemarkus** [comment]
this should partially be fixed with #12917 
/remove-lifecycle rotten
